steam_buddy/streaming.py

```diff
@@ -0,0 +1,84 @@
+import os
+import shlex
+import time
+import subprocess as sp
+
+
+class StreamServer:
+
+    def __init__(self, sls_conf_file):
+        self.sls_conf_file = sls_conf_file
+        self._ffmpeg_cmd = "ffmpeg"
+        self._sls_cmd = "sls"
+        self._sls = None
+        self._ffmpeg = None
+
+    def __generate_sls_conf(self):
+        pass
+
+    def __start_sls(self):
+        if self._sls is None:
+            self._sls = sp.Popen([self._sls_cmd, '-c', self.sls_conf_file])
+        else:
+            raise(Exception("Error starting SLS: Already started"))
+
+    def __stop_sls(self):
+        self._sls.terminate()
+        try:
+            self._sls.wait(5)
+        except sp.TimeoutExpired:
+            self._sls.kill()
+        self._sls = None
+
+    def __start_ffmpeg(self, local=False):
+        command_line = ""
+        VCODEC = "libx264"
+        VCODEC_OPTIONS = "-preset ultrafast -tune zerolatency"
+        VSIZE = "1920x1080"
+        ACODEC = "libmp3lame"
+        ACODEC_OPTIONS = ""
+        FPS = "60"
+
+        if local:
+            STREAM = "screen_" + \
+                    str(time.strftime("%Y%m%d_%H%M%S")) + \
+                    ".mp4"
+        else:
+            STREAM = '"srt://localhost:8080?streamid=uplive.gameros/live/stream"'
+
+        command_line = self._ffmpeg_cmd + " " + \
```

Comment:
This would look better if it was an array. Then you don't need spaces. See this code for an example: https://github.com/gamer-os/steam-buddy/blob/0cc663cc5c991be32a4bb3d2fbaa087d5a96666e/steam_buddy/platforms/flathub.py#L125

Reply:
Thank you very much for taking the time to review it. I would love to do it that way and many earlier tests were made that way. The problem I found is with the codecs options that are a string with many options inside that don't work well with Piper's quotation. Maybe I can do a concatenation that adds spaces and then pass it through shlex as I do here, but the solution is not that simple with ffmpeg unfortunately.